include/libtorrent/add_torrent_params.hpp

```diff
@@ -370,6 +370,11 @@ TORRENT_VERSION_NAMESPACE_3
 		std::time_t last_download = 0;
 		std::time_t last_upload = 0;
 
+		// if this is an SSL torrent, specify the root certificate here.
+		// This is so that an ssl torrent can be added with only an infohash
+		// and certificate.
+		std::string root_certificate;
```

Comment:
unfortunately `add_torrent_params` has some restrictions to maintain ABI compatibility. If you move this field to the end of the class, I believe it's OK